api/lib/infrastructure/events/EventDispatcherLogger.js

```diff
@@ -0,0 +1,47 @@
+class EventDispatcherLogger {
+  constructor(monitoringTools, settings) {
+    this._monitoringTools = monitoringTools;
+    this._settings = settings;
+  }
+  onEventDispatchStarted(event, eventHandlerName) {
+    if (this._settings.logging.enableLogStartingEventDispatch) {
+      this._monitoringTools.logInfoWithCorrelationIds({
+        ..._buildLoggingContext({ event, eventHandlerName }),
+        message: 'EventDispatcher : Event dispatch started.',
+      });
+    }
+  }
+
+  onEventDispatchSuccess(event, eventHandlerName) {
+    if (this._settings.logging.enableLogEndingEventDispatch) {
+      this._monitoringTools.logInfoWithCorrelationIds({
+        ..._buildLoggingContext({ event, eventHandlerName }),
+        message: 'EventDispatcher : Event dispatched successfully',
+      });
+    }
+  }
+
+  onEventDispatchFailure(event, eventHandlerName, error) {
+    if (this._settings.logging.enableLogEndingEventDispatch) {
+      this._monitoringTools.logInfoWithCorrelationIds({
```

Comment:
Suggestion: A faire dans une autre PR peut être !! mais vaut mieux utiliser le logErrorWithCorrelationIds après l'avoir adapter pour afficher les métriques.  Cela permettra de voir les logs avec un niveau ERROR et non INFO

Reply:
J'y ai pensé et j'ai quasi fait le refacto mais j'ai un soucis.
Au final on passe l'objet `error` au log et donc le message du log c'est le message de l'erreur. 
Moi je voudrais que le message du log soit "EventDispatcher : An error occurred while dispatching the event" pour être raccord avec le message de début de dispatch (et aussi contextualiser l'erreur) du coup j'aurais du faire des manip. 
Bref là on a le log + l'alerting et par ailleurs l'erreur remonte dans hapi qui la loggue comme toute autre erreur, donc on est safe.